### DISCUSSION TITLE
Azure Kubernetes Compatability with Strimzi (JDK Versions)

### DISCUSSION THREAD

**Discussion by Sai-Charan-Madhvaraj** [strimzi/strimzi-kafka-operator#8797] [General]
Hi Team, 

We are running Strimzi version 0.27.1 with kafka version 3.0.0, 

The JDK version in above versions are "11.0.13" 

<img width="589" alt="image" src="https://github.com/strimzi/strimzi-kafka-operator/assets/95335222/25562eb6-a5c6-40b5-882b-0a3a2921473c">

According to Azure Kubernetes versions, 1.25 of K8s does not support JDK of 11.0.13 and preferred versions to use are JDK 11.0.16 and later or JDK 15 and later, which fully support cgroup v2

<img width="853" alt="image" src="https://github.com/strimzi/strimzi-kafka-operator/assets/95335222/1cc7d14f-3475-43ff-94d0-f77082cd1760">

https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-components-breaking-changes-by-version

Considering the Azure kubernetes version 1.25, strimzi 0.27 might not be compatible and any other options available to upgrade the JDK with out strimzi/Kafka upgrade. 
Also , do we have the matrix availability of strimzi versions and respective JDK being used ?

**Comment by scholzj** [comment]
I'm not sure what is expected to be the perceived incompatibility. Even Java 8 containers can run without any problem on Kubernetes 1.27. Cgroups v2 is not something that needs special support for the container to work. All that changes there is how it provides information about the available resources. It is supported even in Java/OpenJDK 11 - but I do not remember from which exact patch release, you would need to check it. In any case, you can always upgrade to newer Strimzi versions that use newer Java versions. 

Also, keep in mind that Strimzi 0.27 is not compatible with Kubernetes 1.25 because of Kubernetes API changes unrelated to Java (see the table in https://strimzi.io/downloads/). You need at least Strimzi 0.29! (But preferably, you should upgrade to something much newer than 0.29 of course)
